### PR TITLE
Give pointers to versions more descriptive names

### LINKED ISSUE #4800: Rename RepositoryObject associations
To help make referring to these associations clearer, please rename them as follows:
* "head" -> "last_closed_version"
* "current" -> "head_version"
* "open" -> "opened_version"

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔
Fixes #4800 

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Review by justinlittman — APPROVED** [review]

File: db/structure.sql [hunk 1]
```diff
@@ -828,24 +828,24 @@ CREATE INDEX "index_repository_object_versions_on_structural_isMemberOf" ON publ
...
+CREATE UNIQUE INDEX index_repository_objects_on_external_identifier ON public.repository_objects USING btree (external_identifier);
 
 
 --
--- Name: index_repository_objects_on_external_identifier; Type: INDEX; Schema: public; Owner: -
+-- Name: index_repository_objects_on_head_version_id; Type: INDEX; Schema: public; Owner: -
 --
 
-CREATE UNIQUE INDEX index_repository_objects_on_external_identifier ON public.repository_objects USING btree (external_identifier);
+CREATE INDEX index_repository_objects_on_head_version_id ON public.repository_objects USING btree (head_version_id);
```
Comment (mjgiarlo) [inline db/structure.sql:841]: I'm not sure I grok where these changes are from. @jcoyne @justinlittman do y'all?

Comment (jcoyne) [reply db/structure.sql:841]: Isn't this just an alphabetical reorder of what was already there (after current_id -> head_version_id)?

Comment (mjgiarlo) [reply db/structure.sql:841]: Maybe they got into a different order and GH is rendering the diff in a way that is confusing me.

Comment (mjgiarlo) [reply db/structure.sql:841]: Yes, that's what happened. Resolving!

**Pull request merged**